Mlem/Logic/Participating/Post Comment.swift

```diff
@@ -8,120 +8,53 @@
-{
-    do
-    {
-        let commentPostingCommandResult: String = try await sendPostCommand(appState: appState, account: account, endpoint: "comment", arguments: [
-            "content": commentContents,
-            "post_id": post.id
-        ])
-
-        print("Successfuly posted comment: \(commentPostingCommandResult)")
-
-        if !commentPostingCommandResult.contains("\"error\"")
-        {
-            do
-            {
-                let postedComment: Comment = try await parseComments(commentResponse: commentPostingCommandResult, instanceLink: account.instanceLink).first!
-                
-                withAnimation(.interactiveSpring(response: 0.4, dampingFraction: 1, blendDuration: 0.4))
-                {
-                    commentTracker.comments.prepend(postedComment)
-                }
-            }
-            catch let commentParsingError
-            {
-                
-                appState.alertTitle = "Couldn't read updated comment"
-                appState.alertMessage = "Refresh comments to see your new comment."
-                appState.isShowingAlert.toggle()
-                
-                print("Failed while parsing updated comment: \(commentParsingError)")
-            }
-        }
-        else
-        {
-            print("Received error from server")
-
-            throw CommentPostingFailure.receivedInvalidServerResponse
-        }
-    }
-    catch let commentPostingError
-    {
-        print("Failed while posting comment: \(commentPostingError)")
-
-        throw CommentPostingFailure.couldNotPost
+func postComment(
+    to post: APIPostView,
+    commentContents: String,
+    commentTracker: CommentTracker,
+    account: SavedAccount,
+    appState: AppState
+) async throws {
+    let request = try CreateCommentRequest(
+        account: account,
+        content: commentContents,
+        languageId: nil,
+        parentId: nil,
+        postId: post.id
+    )
+    
+    let response = try await APIClient().perform(request: request)
+    let comment = HierarchicalComment(comment: response.commentView, children: [])
+    withAnimation(.interactiveSpring(response: 0.4, dampingFraction: 1, blendDuration: 0.4)) {
+        commentTracker.comments.prepend(comment)
     }
 }
 
 @MainActor
-func postComment(to comment: Comment, post: Post, commentContents: String, commentTracker: CommentTracker, account: SavedAccount, appState: AppState) async throws
-{
-    
+func postComment(
+    to comment: APICommentView,
+    post: APIPostView,
+    commentContents: String,
+    commentTracker: CommentTracker,
+    account: SavedAccount,
+    appState: AppState
+) async throws {
     let dominantLanguage = NSLinguisticTagger.dominantLanguage(for: commentContents)
+    let request = try CreateCommentRequest(
+        account: account,
+        content: commentContents,
+        languageId: dominantLanguage == "en" ? 37 : nil,
```

Comment:
We should really build up the list of languages... 😅

Reply:
whole heartedly agree - I'd suggest this be treated as a separate task, this one was already in the implementation so I kept it, but mapping between the API value and Apple's values would ideally be moved into an extension or such so we can use it throughout when required

Reply:
Maybe add a todo comment, so we know to return to it.

Reply:
added in a74f57900e668f3850afdfbcadd0a08a24dcb202